### PR TITLE
properly shutdown litserve workers

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
A test with a side-effect that raises an exception and asserts terminate is called would be useful here

File: src/litserve/server.py [hunk 1]
```diff
@@ -717,15 +717,16 @@ def run(
...
+        try:
+            servers = self._start_server(port, num_uvicorn_servers, log_level, sockets, uvicorn_worker_type)
+            for s in servers:
+                s.join()
+        finally:
+            print("Shutting down LitServe")
+            for w in litserve_workers:
+                w.terminate()
+                w.join()
+            manager.shutdown()
```
Comment (Borda) [inline src/litserve/server.py:729]: why the finally is needed?

Comment (aniketmaurya) [reply src/litserve/server.py:729]: so that we always make sure that the LitServer workers are terminated after terminating the uvicorn server

**Comment by aniketmaurya** [comment]
thanks for the feedback @awaelchli! added a test to assert shutdown is called.

File: src/litserve/server.py (hunk 1 above)
Comment (Borda) [reply src/litserve/server.py:729]: so in fact, you do?
```py
try:
    servers = self._start_server(port, num_uvicorn_servers, log_level, sockets, uvicorn_worker_type)
    for s in servers:
        s.join()
    print("Shutting down LitServe")
    for w in litserve_workers:
        w.terminate()
        w.join()
    manager.shutdown()
except:
    pass
```

**Pull request merged**